provider/diag/diagnostics.go

```diff
@@ -15,7 +15,10 @@ type Diagnostics []Diagnostic
 
 var _ sort.Interface = (*Diagnostics)(nil)
 
-func (diags Diagnostics) Error() string {
+// Deprecated! Only here for backward compatibility (specifically in the gRPC protocol). Do not use unless
+//  you are 100% sure you need this.
+// Converts a list of diagnostics into an error.
+func (diags Diagnostics) DeprecatedToError() string {
```

Comment:
```suggestion
func (diags Diagnostics) ToError() string {
```
Would just mark it as `Deprecated:` as described in [here](https://stackoverflow.com/questions/7849663/how-do-you-mark-code-as-deprecated-in-go) (The `Deprecated!` usage above is also non standard)